### ISSUE TITLE
fmiComponentEnvironment argument in allocateMemory and freeMemory callbacks

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/fmi-standard#270]
**Reported by masoud on 11 Jul 2014 08:24 UTC**
In FMI2.0, the arguments of the logger callback function has changed and the a new argument “fmiComponentEnvironment” has been added. This is very useful and prevents the simulator from using global variables to pass user-data information into the logger.
Would be possible to do the same for allocateMemory and freeMemory callback functions.
 A very simple usage of this new argument would be to count the allocated/freed memory blocks and warn the user about memory leak in the FMU block. An advanced usage would be to develop a memory management mechanism inside the FMU block to reduce the burden of freeing allocated memory. Currently we use global variables to pass environment data into the malloc/calloc/free functions.

----

Migrated-From: https://trac.fmi-standard.org/ticket/270

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 11 Jul 2014 09:24 UTC**
Discussion in the web-meeting:
There is agreement that this would be useful (so that the environment can identify the storage allocated from a specific FMU). However, such a change is too drastic at the current stage (a few days before the release and with many prototypes available) because the C-interface would need to be changed.

One (ugly) workaround proposed by Andreas Junghanns: Pass different function pointers to every FMU (like alloc1, alloc2, alloc3, ..., that are precompiled in the environment).

To summarize, the ticket is closed.
